src/Diagnostics.RuntimeHost/Services/HealthCheckService/HealthCheckService.cs

```
@@ -0,0 +1,74 @@
+using System;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Configuration;
+
+namespace Diagnostics.RuntimeHost.Services
+{
+    public interface IHealthCheckService : IDisposable
+    {
+        Task RunHealthCheck(bool retry);
+    }
+
+    public class HealthCheckService : IHealthCheckService
+    {
+        private string OutboundConnectivityCheckUrl;
+        private HttpClient _httpClient;
+        IConfiguration _configuration;
+        bool IsOutboundConnectivityCheckEnabled = false;
+
+        public HealthCheckService(IConfiguration Configuration)
+        {
+            _configuration = Configuration;
+            IsOutboundConnectivityCheckEnabled = Convert.ToBoolean(_configuration["HealthCheckSettings:IsOutboundConnectivityCheckEnabled"]);
+            if (IsOutboundConnectivityCheckEnabled)
+            {
+                OutboundConnectivityCheckUrl = _configuration["HealthCheckSettings:OutboundConnectivityCheckUrl"];
+                if (OutboundConnectivityCheckUrl != null)
```

if null, should we throw a invalid config exception

---

also check or empty or null